python/paddle/fluid/tests/unittests/test_corr.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.fluid as fluid
+import unittest
+import numpy as np
+import six
+import paddle
+import warnings
+
+
+def numpy_corr(np_arr, rowvar=True, dtype='float64'):
+    return np.corrcoef(np_arr, rowvar=rowvar, dtype=dtype)
+
+
+class Corr_Test(unittest.TestCase):
+    def setUp(self):
+        self.shape = [20, 10]
+
+    def test_tensor_corr_default(self):
+        typelist = ['float64', 'float32']
+        places = [fluid.CPUPlace()]
+        if fluid.core.is_compiled_with_cuda():
+            places.append(fluid.CUDAPlace(0))
+        for idx, p in enumerate(places):
+            if idx == 0:
+                paddle.set_device('cpu')
+            else:
+                paddle.set_device('gpu')
+
+            for dtype in typelist:
+                np_arr = np.random.rand(*self.shape).astype(dtype)
+                tensor = paddle.to_tensor(np_arr, place=p)
+                corr = paddle.linalg.corrcoef(tensor)
+                np_corr = numpy_corr(np_arr, rowvar=True, dtype=dtype)
+                self.assertTrue(np.allclose(np_corr, corr.numpy(), atol=1.e-2))
```

Comment:
commonly, we set atol=0, and use rtol=1.e-6 to check precision, atol=1.e-2 is too large to check error.
if we do have significant difference with numpy, should explain reason and fix error.